golem-worker-executor-base/src/worker.rs

```diff
@@ -86,6 +89,41 @@ pub struct Worker<Ctx: WorkerCtx> {
     oom_retry_config: RetryConfig,
 }
 
+impl<Ctx: WorkerCtx> Worker<Ctx> {
+    pub async fn get_component_metadata(
+        worker: &Arc<Worker<Ctx>>,
```

Comment:
This should be a regular method, I don't see why it would take an `Arc<Worker<Ctx>>` (just take `&self`)